truffle/src/main/java/org/jruby/truffle/core/array/ArrayNodes.java

```diff
@@ -911,6 +916,56 @@ protected Object fillFallback(VirtualFrame frame, DynamicObject array, Object[]
 
     }
 
+    @CoreMethod(names = "hash_internal")
+    public abstract static class HashNode extends ArrayCoreMethodNode {
+
+        @Child private ToIntNode toIntNode;
+
+        @Specialization(guards = "isNullArray(array)")
+        public long hashNull(VirtualFrame frame, DynamicObject array) {
```

Comment:
When the frame is not needed, it's usually best to avoid it in the arguments for clarity.